cdc/sink/buffer_sink.go

```diff
@@ -191,8 +193,13 @@ func (b *bufferSink) FlushRowChangedEvents(ctx context.Context, tableID model.Ta
 		tableID:    tableID,
 		resolvedTs: resolvedTs,
 	}:
+		return b.getTableCheckpointTs(tableID), nil
+	default:
 	}
-	return b.getTableCheckpointTs(tableID), nil
+	// since n.checkpoints is refreshed asynchronously in a background goroutine,
+	// we return a n.checkpoints every time FlushRowChangedEvents be called
+	// the caller will always check the returned checkpointTs
+	return b.getTableCheckpointTs(tableID), cerrors.ErrFlushTsBlocking.FastGenByArgs()
```

Comment:
```go
	default:
		// since n.checkpoints is refreshed asynchronously in a background goroutine,
		// we return a n.checkpoints every time FlushRowChangedEvents be called
		// the caller will always check the returned checkpointTs
		return b.getTableCheckpointTs(tableID), cerrors.ErrFlushTsBlocking.FastGenByArgs()
	}
```

I prefer returning blocking error in the default case.

Reply:
By dropping flushMsg, will it cause flowcontrol deadlock? @liuzix

Reply:
> By dropping flushMsg, will it cause flowcontrol deadlock? @liuzix

The deadlock issue will fix in #4084, after it be merged, dropping flushMsg here will not cause flowcontrol deadlock.